// See README.txt for information and build instructions.

package tutorial;

option java_package = "com.example.tutorial";
option java_outer_classname = "AddressBookProtos";

message Person {
  required string name = 1;
  required int32 id = 2;        // Unique ID number for this person.
  optional string email = 3;
  optional double double_var = 10;
  optional float float_var = 11;
  optional int64 int64_var = 12;
  optional uint32 uint32_var = 13;
  optional uint64 uint64_var = 14;
  optional sint32 sint32_var = 15;
  optional sint64 sint64_var = 16;
  optional fixed32 fixed32_var = 17;
  optional fixed64 fixed64_var = 18;
  optional sfixed32 sfixed32_var = 19;
  optional sfixed64 sfixed64_var = 20;
  optional bool bool_var = 21;
  optional string string_var = 22;
  optional bytes bytes_var = 23;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phone = 4;
}

// Our address book file is just one of these.
message AddressBook {
  repeated Person person = 1;
}
